AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -228,17 +234,41 @@ - (void)requestInstallInformationWith:(NSString *)releaseHash {
         MSLogDebug([MSDistribute logTag],
                    @"Re-attempting in-app updates setup and cleaning up failure info from storage.");
         [MS_USER_DEFAULTS removeObjectForKey:kMSUpdateSetupFailedPackageHashKey];
+        [MS_USER_DEFAULTS removeObjectForKey:kMSTesterAppUpdateSetupFailedKey];
       }
     }
+    
+    // Create the request ID string and persist it.
+    NSString *requestId = MS_UUID_STRING;
+    [MS_USER_DEFAULTS setObject:requestId forKey:kMSUpdateTokenRequestIdKey];
 
-    NSURL *url;
     MSLogInfo([MSDistribute logTag], @"Request information of initial installation.");
-
-    // Most failures here require an app update. Thus, it will be retried only on next App instance.
-    url = [self buildTokenRequestURLWithAppSecret:self.appSecret releaseHash:releaseHash];
-    if (url) {
-      [self openUrlInAuthenticationSessionOrSafari:url];
-    }
+    
+    // Use swizzling to get [UIApplication sharedApplication]
+    SEL sharedAppSel = NSSelectorFromString(@"sharedApplication");
```

Comment:
`MSUtility+Application` already contains logic for this. Why don't we expose the method so we can use that across the app?